### ISSUE TITLE
Loss function in seq2seq model

### ISSUE THREAD

**Issue by NTT123** [pytorch/tutorials#325]
https://github.com/pytorch/tutorials/blob/cd4953d503eaf7194092a9f8945e54f6b30a909a/intermediate_source/seq2seq_translation_tutorial.py#L590

If I understand it correctly, the loss for each pair (input,target) in this model is actually the negative log-likelihood of the target sequence given the input sequence which is the sum of log-likelihoods according to the chain rule.

Meanwhile, this line returns the average therefore implementing a different loss function for which a very long target sequence contributes the same among of *negative log-likelihood*  loss as of a very short sequence.
